### DISCUSSION TITLE
Program has more than one entry point defined. Compile with /main to specify the type that contains the entry point.

### DISCUSSION THREAD

**Discussion by Am199503** [QuestPDF/QuestPDF#140] [Q&A] [answered]
Hi
i am trying to print pdf but it's giving me issue,
Please help

`using QuestPDF.Drawing;
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;
using System.Diagnostics;

namespace ErpOneCore.Controllers
{
   // #region constructor
    [SessionTimeout]

public class InvoiceDocument : IDocument
{
    
    /* code omitted */
     public InvoiceModel Model { get; }
     static  private QuestRepository QuestRepository;

    public InvoiceDocument(InvoiceModel model)
    {
        Model = model;
    }

    public DocumentMetadata GetMetadata() => DocumentMetadata.Default;

    public void Compose(IDocumentContainer container)
    {
        container
            .Page(page =>
            {
                page.Margin(50);
            
                page.Header().Element(ComposeHeader);
                page.Content().Element(ComposeContent);

                    
                page.Footer().AlignCenter().Text(x =>
                {
                    x.CurrentPageNumber();
                    x.Span(" / ");
                    x.TotalPages();
                });
            });
    }

    void ComposeHeader(IContainer container)
    {
        var titleStyle = TextStyle.Default.Size(20).SemiBold().Color(Colors.Blue.Medium);
    
        container.Row(row =>
        {
            row.RelativeItem().Column(column =>
            {
                column.Item().Text($"Invoice #{"12000"}", titleStyle);

                column.Item().Text(text =>
                {
                    text.Span("Issue date: ", TextStyle.Default.SemiBold());
                    text.Span($"{"4/5/22":d}");
                });
                
                column.Item().Text(text =>
                {
                    text.Span("Due date: ", TextStyle.Default.SemiBold());
                    text.Span($"{"4/4/20":d}");
                });
            });

            row.ConstantItem(100).Height(50).Placeholder();
        });
    }

    void ComposeContent(IContainer container)
    {
        container
            .PaddingVertical(40)
            .Height(250)
            .Background(Colors.Grey.Lighten3)
            .AlignCenter()
            .AlignMiddle()
            .Text("Content", TextStyle.Default.Size(16));
    }
     static void Main(string[] args)
 {
     var filePath = "invoice.pdf";

    var model =  QuestRepository.detailsq();
    var document = new InvoiceDocument(model);
     document.GeneratePdf(filePath);

    Process.Start("explorer.exe", filePath);
 }
}
}`

![Screenshot 2022-03-11 194907](https://user-images.githubusercontent.com/101409683/157885451-d68c7f2b-48ed-4d00-9636-6ecafe220e18.png)

**Comment by girlpunk** [comment, marked as answer]
You have another class in the project that defines `static void Main(string[] args)`. This isn't a QuestPDF error, but comes from the c# language itself. See Microsoft's documentation on [the error](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0017) and [the `Main()` function in general](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/program-structure/main-command-line) for more information.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
